src/plugins/data/public/search/fetch/extract_warnings.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { estypes } from '@elastic/elasticsearch';
+import { i18n } from '@kbn/i18n';
+import { SearchResponseWarning } from '../types';
+
+/**
+ * @internal
+ */
+export function extractWarnings(
```

Comment:
This function doesn't fit well in inspector currently.

1. Inspector is just an interface, we would have to use the `adapter` field, which is RequestAdapter, which is very generic
2. Doing this changes means we need to import `IKibanaSearchResponse` from the `data` plugin, whereas currently the Inspector plugin has no awareness of ES.

Reply:
RequestAdapter already has `getRequests` method, I tought we could add another method `getWarnings` there, so consumer would use it as `inspector.requests.getWarnings()`

i don't see IKibanaSearchResponse used anywhere in this file ? i think we only need to depend on @kbn/es types

Reply:
> RequestAdapter already has getRequests method, I tought we could add another method getWarnings there, so consumer would use it as inspector.requests.getWarnings()

If we're looping through the warnings instead of the requests, we don't have enough context to for the ShardFailureOpenModalButton component, which requires more than just the warning. So I think using `inspector.adapter.getRequests` when handling the requests is the right way to go, rather than refactoring to make things warning-centric. 

> i don't see IKibanaSearchResponse used anywhere in this file ? i think we only need to depend on @kbn/es types

sorry, I think you are right - we would only need estypes.SearchResponse here.